### PR TITLE
Raise event for stack topology change 

### PR DESCRIPTION
**Pull request by anurag6**
Including file write for now until there's code to handle the events.

### CODE REVIEW

File: faucet/valve.py [hunk 1]
```diff
@@ -612,6 +616,15 @@ def _update_stack_link_state(self, ports, now, other_valves):
...
                 for port in valve.dp.stack_ports:
                     ofmsgs_by_valve[valve].extend(valve.host_manager.del_port(port))
+
+                if valve.flood_manager.graph and not graph_updated:
+                    with open('/etc/faucet/dp_graph.json', 'w') as outfile:
+                        outfile.write(json.dumps(networkx.json_graph.node_link_data(valve.flood_manager.graph), default=str))
+                    self._notify(
+                            {'STACK_TOPO_CHANGE': {
+                                'stack_root': valve.dp.stack_root_name,
+                                'graph': json.dumps(networkx.json_graph.node_link_data(valve.flood_manager.graph), default=str)}})
```
Comment (grafnu) [inline faucet/valve.py]: Why encode graph as string?  At this point, I think we should be sending through the actual data (which then gets encoded as json by notify)...

File: faucet/valve.py [hunk 2]
```diff
@@ -612,6 +616,15 @@ def _update_stack_link_state(self, ports, now, other_valves):
                     ofmsgs_by_valve[valve].extend(valve.flood_manager.add_vlan(vlan))
                 for port in valve.dp.stack_ports:
                     ofmsgs_by_valve[valve].extend(valve.host_manager.del_port(port))
+
+                if valve.flood_manager.graph and not graph_updated:
+                    with open('/etc/faucet/dp_graph.json', 'w') as outfile:
```
Comment (grafnu) [inline faucet/valve.py]: We should ditch this as soon as we can, so we don't inadvertently rely on it!

File: faucet/valve.py [hunk 3]
```diff
@@ -601,8 +603,10 @@ def _update_stack_link_state(self, ports, now, other_valves):
                     port_stack_up = port.is_stack_up()
                     for valve in stacked_valves:
                         valve.flood_manager.update_stack_topo(port_stack_up, self.dp, port)
+
         if stack_changes:
             self.logger.info('%u stack ports changed state' % stack_changes)
+            graph_updated = False
```
Comment (grafnu) [inline faucet/valve.py]: graph_exported?  Not sure this variable is named accurately.

File: faucet/valve.py [hunk 4]
```diff
@@ -612,6 +616,15 @@ def _update_stack_link_state(self, ports, now, other_valves):
                     ofmsgs_by_valve[valve].extend(valve.flood_manager.add_vlan(vlan))
                 for port in valve.dp.stack_ports:
                     ofmsgs_by_valve[valve].extend(valve.host_manager.del_port(port))
+
+                if valve.flood_manager.graph and not graph_updated:
```
Comment (grafnu) [inline faucet/valve.py]: Why is this here -- can this code be placed outside of the for loop?   Are there flood_manages somehow that don't have a graph attached? Would be a lot cleaner conceptually...

**Review by grafnu — CHANGES_REQUESTED** [review]

File: faucet/valve.py (hunk 2 above)
Comment (anurag6) [reply faucet/valve.py]: Will remove it once the daq changes are ready.

File: faucet/valve.py (hunk 4 above)
Comment (anurag6) [reply faucet/valve.py]: A cleaner way wit hverification might be to have a separate loop, where we iterate until we find a flood manager with the graph and print it and break. Would that make sense?

**Comment by grafnu** [comment]
Yah -- just trying to avoid the conditional logic bit, it always ends up
just being more confusing in the long run. I think the loop-and-break would
be most understandable.

**Review by grafnu — APPROVED** [review]
